lib/timetable-visualisation/Visualisation.tsx

```diff
@@ -0,0 +1,134 @@
+import { useCallback, useEffect, useState } from "react";
+import csvToJson from "csvtojson";
+import { dateToDefaultLocalDateString, loadCSV } from "../utils/timetable";
+import { DevelopmentPlan } from "../types/timetable";
+
+import "govuk-frontend/dist/govuk/govuk-frontend.min.css";
+
+const DEFAULT_TIMETABLE_DATA: DevelopmentPlan = {
+  reference: "",
+  name: "",
+  description: "",
+  developmentPlanType: "",
+  periodStartDate: new Date(),
+  periodEndDate: new Date(),
+  developmentPlanGeography: "",
+  documentationUrl: "",
+  adoptedDate: new Date(),
+  organisations: [],
+  entryDate: new Date(),
+  startDate: new Date(),
+  endDate: new Date(),
+  timetableEvents: [],
+};
+
+type VisualisationProps = {
+  stagesFilepath: string;
+  headersFilepath: string;
+};
+
+export const Visualisation = (props: VisualisationProps) => {
+  const { stagesFilepath, headersFilepath } = props;
+  const [timetableData, setTimetableData] = useState<DevelopmentPlan>(
+    DEFAULT_TIMETABLE_DATA
+  );
+  const loadData = useCallback(async () => {
+    let loadedData: DevelopmentPlan = DEFAULT_TIMETABLE_DATA;
+    await loadCSV(stagesFilepath).then(async (data) => {
+      await csvToJson()
+        .fromString(data)
+        .then((jsonObj) => {
+          loadedData = {
+            ...loadedData,
+            timetableEvents: jsonObj,
+          };
+        });
+    });
+    await loadCSV(headersFilepath).then(async (data) => {
+      await csvToJson()
+        .fromString(data)
+        .then((jsonObj) => {
+          loadedData = {
+            ...loadedData,
+            ...jsonObj[0],
+          };
+        });
+    });
+    setTimetableData(loadedData);
```

Comment:
I don't think we want to be mixing `await` and `.then`, we can simplify this function:

```suggestion
    const stagesData = await loadCSV(stagesFilepath);
    const events = await csvToJson().fromString(stagesData);

    const headersData = await loadCSV(headersFilepath);
    const headers = await csvToJson().fromString(headersData);

    const loadedData: DevelopmentPlan = {
      ...DEFAULT_TIMETABLE_DATA,
      timetableEvents: events,
      ...headers[0],
    };

    setTimetableData(loadedData);
```

Reply:
Thanks Ronan :) I lifted this part from plan-making code and didn't realise we were mixing .then()s and awaits

Reply:
change applied 👍